pkgs/by-name/js/jsonschema-cli/package.nix

```diff
@@ -0,0 +1,66 @@
+{
+  lib,
+  fetchCrate,
+  rustPlatform,
+  versionCheckHook,
+  nix-update-script,
+  runCommand,
+  testers,
+  jsonschema-cli,
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "jsonschema-cli";
+  version = "0.29.0";
+
+  src = fetchCrate {
+    inherit pname version;
+    hash = "sha256-kpdvvCnMsHfogXmAqNeo1Cl1hZtCPHqkfhYm8ipWToo=";
+  };
+
+  useFetchCargoVendor = true;
+  cargoHash = "sha256-Yti1KKJDRXMhDo84/ymZk2AkWp9HtU2LW2h63gfzIGY=";
+
+  nativeInstallCheckInputs = [
+    versionCheckHook
+  ];
+  doInstallCheck = true;
+  versionCheckProgram = "${placeholder "out"}/bin/${meta.mainProgram}";
```

Comment:
I think it is discouraged to use mainProgram here. The version check works without this line.

Reply:
https://github.com/NixOS/nixpkgs/blob/6d0e5aea88d0f8fa0d0e1258c65c4f799c46717e/doc/hooks/versionCheckHook.section.md?plain=1#L34


The default depends on `pname`. It sounds robust to me to specify `meta.mainProgram`.